### ISSUE TITLE
MySQL >= 8 - Parse error on column with 'expression default values'

### ISSUE THREAD

**Issue by Cielquan** [sqlfluff/sqlfluff#5204]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a SQL script where I create a table, which has a column of type `date` and should get the current date as default. Therefore I wrote the following which results in an parse error:

```sql
create table `tickets` (
    `id` serial primary key,
    `material_number` varchar(255) default null,
    `material_name` varchar(255) default null,
    `date_created` date not null default (current_date),
    `date_closed` date default null
);
```

When I remove the parenthesis around `current_date` the parse error disappears.

Since MySQL 8 parenthesis are required around expression default values.
See: https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html

### Expected Behaviour

The parenthesis around `current_date` should not result in a parse error.

### Observed Behaviour

A parse error occured:

```console
$ sqlfluff lint --verbose --ignore-local-config --dialect mysql test.sql 
==== sqlfluff ====
sqlfluff:                2.3.1 python:                3.10.12
implementation:        cpython verbosity:                   1
dialect:                 mysql templater:               jinja
rules:                                all
==== readout ====

== [test.sql] FAIL                                                                                                                                                     
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'create
                       | table `tickets` (\n    `id` serial...'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

When I run the parse command I get this:
```console
$ sqlfluff parse --verbose --ignore-local-config --dialect mysql test.sql 
==== sqlfluff ====
sqlfluff:                2.3.1 python:                3.10.12
implementation:        cpython verbosity:                   1
dialect:                 mysql templater:               jinja
rules:                                all
=== [ path: test.sql ] ===

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'statement'
[L:  1, P:  1]      |            code:                                             'create'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            code:                                             'table'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            back_quote:                                       '`tickets`'
[L:  1, P: 23]      |            whitespace:                                       ' '
[L:  1, P: 24]      |            bracketed:
[L:  1, P: 24]      |                start_bracket:                                '('
[L:  1, P: 25]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                back_quote:                                   '`id`'
[L:  2, P:  9]      |                whitespace:                                   ' '
[L:  2, P: 10]      |                code:                                         'serial'
[L:  2, P: 16]      |                whitespace:                                   ' '
[L:  2, P: 17]      |                code:                                         'primary'
[L:  2, P: 24]      |                whitespace:                                   ' '
[L:  2, P: 25]      |                code:                                         'key'
[L:  2, P: 28]      |                comma:                                        ','
[L:  2, P: 29]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                back_quote:                                   '`material_number`'
[L:  3, P: 22]      |                whitespace:                                   ' '
[L:  3, P: 23]      |                code:                                         'varchar'
[L:  3, P: 30]      |                bracketed:
[L:  3, P: 30]      |                    start_bracket:                            '('
[L:  3, P: 31]      |                    numeric_literal:                          '255'
[L:  3, P: 34]      |                    end_bracket:                              ')'
[L:  3, P: 35]      |                whitespace:                                   ' '
[L:  3, P: 36]      |                code:                                         'default'
[L:  3, P: 43]      |                whitespace:                                   ' '
[L:  3, P: 44]      |                code:                                         'null'
[L:  3, P: 48]      |                comma:                                        ','
[L:  3, P: 49]      |                newline:                                      '\n'
[L:  4, P:  1]      |                whitespace:                                   '    '
[L:  4, P:  5]      |                back_quote:                                   '`material_name`'
[L:  4, P: 20]      |                whitespace:                                   ' '
[L:  4, P: 21]      |                code:                                         'varchar'
[L:  4, P: 28]      |                bracketed:
[L:  4, P: 28]      |                    start_bracket:                            '('
[L:  4, P: 29]      |                    numeric_literal:                          '255'
[L:  4, P: 32]      |                    end_bracket:                              ')'
[L:  4, P: 33]      |                whitespace:                                   ' '
[L:  4, P: 34]      |                code:                                         'default'
[L:  4, P: 41]      |                whitespace:                                   ' '
[L:  4, P: 42]      |                code:                                         'null'
[L:  4, P: 46]      |                comma:                                        ','
[L:  4, P: 47]      |                newline:                                      '\n'
[L:  5, P:  1]      |                whitespace:                                   '    '
[L:  5, P:  5]      |                back_quote:                                   '`date_created`'
[L:  5, P: 19]      |                whitespace:                                   ' '
[L:  5, P: 20]      |                code:                                         'date'
[L:  5, P: 24]      |                whitespace:                                   ' '
[L:  5, P: 25]      |                code:                                         'not'
[L:  5, P: 28]      |                whitespace:                                   ' '
[L:  5, P: 29]      |                code:                                         'null'
[L:  5, P: 33]      |                whitespace:                                   ' '
[L:  5, P: 34]      |                code:                                         'default'
[L:  5, P: 41]      |                whitespace:                                   ' '
[L:  5, P: 42]      |                bracketed:
[L:  5, P: 42]      |                    start_bracket:                            '('
[L:  5, P: 43]      |                    code:                                     'current_date'
[L:  5, P: 55]      |                    end_bracket:                              ')'
[L:  5, P: 56]      |                comma:                                        ','
[L:  5, P: 57]      |                newline:                                      '\n'
[L:  6, P:  1]      |                whitespace:                                   '    '
[L:  6, P:  5]      |                back_quote:                                   '`date_closed`'
[L:  6, P: 18]      |                whitespace:                                   ' '
[L:  6, P: 19]      |                code:                                         'date'
[L:  6, P: 23]      |                whitespace:                                   ' '
[L:  6, P: 24]      |                code:                                         'default'
[L:  6, P: 31]      |                whitespace:                                   ' '
[L:  6, P: 32]      |                code:                                         'null'
[L:  6, P: 36]      |                newline:                                      '\n'
[L:  7, P:  1]      |                end_bracket:                                  ')'
[L:  7, P:  2]      |    statement_terminator:                                     ';'
[L:  7, P:  3]      |    newline:                                                  '\n'
[L:  8, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'create table
                       | `tickets` (\n    `id` serial...'
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
```

### How to reproduce

- Create a file `test.sql` with the content of the example above
- Run `sqlfluff parse --verbose --ignore-local-config --dialect mysql test.sql`
- See the error

### Dialect

mysql

### Version

sqlfluff 2.3.1
Python 3.10.12

installed via pipx

### Configuration

Default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Cielquan** [comment]
Thanks for the fast fix :)
